src/Microsoft.Azure.WebJobs.Extensions.EventHubs/EventHubConfiguration.cs

```diff
@@ -437,6 +439,13 @@ private static Task<object> ConvertPocoToEventData(object arg, Attribute attrRes
             return Task.FromResult<object>(ConvertString2EventData(JsonConvert.SerializeObject(arg)));
         }
 
+        internal Action<ExceptionReceivedEventArgs> ExceptionHandler { get; set; }
+
+        private void ExceptionReceivedHandler(ExceptionReceivedEventArgs args)
```

Comment:
This somewhat strange level of indirection is necessary because the options class is constructed (and handler wired up) before initialize runs and can configure the actual handler. The handler is a ctor param of EventProcessorOptions so must be set up early.

Note that in the v1 version of the EH library the handler is an **event** so we don't have to do this - we just add an event handler.